Sources/Build/BuildPlan.swift

```diff
@@ -1989,8 +1989,8 @@ public class BuildPlan {
 
         var productMap: [ResolvedProduct: ProductBuildDescription] = [:]
         // Create product description for each product we have in the package graph except
-        // for automatic libraries and plugins, because they don't produce any output.
-        for product in graph.allProducts where product.type != .library(.automatic) && product.type != .plugin {
+        // for automatic libraries, plugins or products which consist solely of binary targets, because they don't produce any output.
+        for product in graph.allProducts where product.type != .library(.automatic) && product.type != .plugin && !product.targets.filter({ !($0.underlyingTarget is BinaryTarget) }).isEmpty {
```

Comment:
Somewhat interesting: binary-only libraries just happened to work because they're also automatic products :)

Reply:
nit: since the condition here is starting to become complex, maybe a private extension on product that makes it more readable